extensions/parquet/compression/src/main/java/io/deephaven/parquet/compress/codec/ZstdCodec.java

```diff
@@ -3,11 +3,48 @@
  */
 package io.deephaven.parquet.compress.codec;
 
+import com.github.luben.zstd.BufferPool;
+import com.github.luben.zstd.NoPool;
+import com.github.luben.zstd.RecyclingBufferPool;
+import io.deephaven.parquet.compress.codec.zstd.ZstdCompressorStream;
+import io.deephaven.parquet.compress.codec.zstd.ZstdDecompressorStream;
+import org.apache.hadoop.io.compress.CompressionInputStream;
+import org.apache.hadoop.io.compress.CompressionOutputStream;
 import org.apache.parquet.hadoop.codec.ZstandardCodec;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
 /**
- * Provides an alternative codec name of "ZSTD" instead of the superclass's "ZSTANDARD".
+ * Provides an alternative codec name of "ZSTD" instead of the superclass's "ZSTANDARD". These streams are also modified
+ * to use the "no finalizer" variant of the underlying streams, so that GC picking up the streams doesn't close the
+ * underlying file.
  */
 public class ZstdCodec extends ZstandardCodec {
+    @Override
+    public CompressionInputStream createInputStream(InputStream stream) throws IOException {
```

Comment:
I think all of these changes belong in a differently-licensed module. Probably `ParquetHadoop`.

Reply:
Or we can make this module apache as well? (If it makes more sense to put into ParquetHadoop b/c they are intimately related, then I agree that might be best.)

Reply:
I moved the code for the moment anyway, but if we end up adding much more in this compression stuff I think it could be nice to open that up, offer as a different option than the upstream CompressionCodecFactory(s).